scalafix-reflect/src/main/scala-3/scalafix/internal/reflect/RuleCompiler.scala

```diff
@@ -0,0 +1,93 @@
+package scalafix.internal.reflect
+
+import scalafix.internal.reflect.exception.CompilationException
+
+import java.io.File
+import java.nio.file.Paths
+import scala.meta.io.AbsolutePath
+
+import dotty.tools.dotc.Compiler
+import dotty.tools.dotc.Run
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.core.Contexts.NoContext.given_Context
+import dotty.tools.dotc.reporting.StoreReporter
+
+import scala.reflect.io.VirtualDirectory
+import scala.reflect.io.AbstractFile
+import scala.reflect.internal.util.AbstractFileClassLoader
+
+import metaconfig.Configured
+import metaconfig.Input
+
+class RuleCompiler(
+    classpath: String,
+    target: AbstractFile = new VirtualDirectory("(memory)", None)
+) {
+  // we are not configuring any settings!
+  // private val settings = new Settings()
+  // settings.deprecation.value = true // enable detailed deprecation warnings
+  // settings.unchecked.value = true // enable detailed unchecked warnings
+  // settings.outputDirs.setSingleOutput(target)
+  // settings.classpath.value = classpath
+
+  private lazy val reporter: StoreReporter = new StoreReporter()
+  private val compiler: Compiler = new Compiler()
+  private val classLoader: AbstractFileClassLoader =
+    new AbstractFileClassLoader(target, this.getClass.getClassLoader)
+
+  @throws(classOf[CompilationException])
```

Comment:
here I'm changing the signature of `compile` by declaring  `throw CompilationException`, is it ok?

Reply:
There are not checked exceptions in scala so I am not sure this will change the signature - did you check?

But anyway, why throwing an exception when you have a `metaconfig.Configured` allowing you to capture errors?

Reply:
the main reason why I'm throwing directly an exception is because we discussed that in case of compilation errors we want to inform the user with a message. I probably misunderstood how we should send the message. 
I'll change this and use `metaconfig.Configured` with this message. Then the exception will be gone as well